core/includes/database/database.inc

```diff
@@ -314,7 +314,25 @@ abstract class DatabaseConnection {
    */
   protected $escapedAliases = array();
 
-  function __construct($dsn, $username, $password, $driver_options = array()) {
+  /**
+   * Creates a database connection object.
+   *
+   * Parameters for creating a DatabaseConnection are identical to that of
+   * creating a PDO object.
+   *
+   * @link https://www.php.net/manual/en/pdo.construct.php
+   *
+   * @param string $dsn
+   *   The Data Source Name, or DSN, contains the information required to
+   *   connect to the database.
+   * @param string $username
+   *   The user name for the DSN string.
+   * @param string $password
+   *   The password for the DSN string.
+   * @param array $driver_options
+   *   A key => value array of driver-specific connection options.
+   */
+  public function __construct($dsn, $username, $password, $driver_options = array()) {
```

Comment:
Type-hinting the `array` left off here and in several other places because parent/child classes would also be required to add the type hint.

Reply:
`// phpcs:ignore` might help in this case.

I played with a more specific `// phpcs:ignore Backdrop.Commenting.FunctionComment.TypeHintMissing`, but then other rules strike (comment too long, ...).

So, maybe just stick with "ignored by humans". :shrug:

Reply:
Yeah I couldn't find a good solution for putting `phpcs` inside the docblock. Outside the docblock (right above the function) it causes other errors and would break PHP's `reflection` API.

Reply:
Reading the docs again... there might be a way.

```
public function __construct($dsn, $username, $password, $driver_options = array()) {// phpcs:ignore
```
^^ But we have no clue, why and adding Backdrop.Commenting.FunctionComment.TypeHintMissing makes the comment too long, but...
```
public function __construct($dsn, $username, $password, $driver_options = array()) {// phpcs:ignore -- Cannot add type hint in base class.
```
^^ Shorter comment, separated with double hyphen seems to work. With a hint, why we ignore this line.

> ...if the comment is placed on a line that also contains code, only that line will be ignored:

Reply:
Thanks, I'll give this a shot and see if it works!

Reply:
This worked great! I added an ignore rule in about 20 places.